deepchecks/vision/vision_data/vision_data.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing the VisionData class and its functions."""
+import typing as t
+from collections import defaultdict
+
+from deepchecks.core.errors import ValidationError
+from deepchecks.utils.logger import get_logger
+from deepchecks.vision.vision_data import TaskType
+from deepchecks.vision.vision_data.format_validators import (validate_additional_data_format,
+                                                             validate_embeddings_format,
+                                                             validate_image_identifiers_format, validate_images_format,
+                                                             validate_labels_format, validate_predictions_format)
+from deepchecks.vision.vision_data.utils import (BatchOutputFormat, get_class_ids_from_numpy_labels,
+                                                 get_class_ids_from_numpy_preds, shuffle_dynamic_loader)
+
+VD = t.TypeVar('VD', bound='VisionData')
+
+
+class VisionData:
+    """VisionData is the base data object of deepcheck vision used as input to checks and suites.
+
+    Vision data at its core is a wrapper of a dynamic loader (such as Pytorch DataLoader or tf Dataset) with additional
+    properties and converters to deepchecks accepted format..
+
+    Parameters
+    ----------
+    dynamic_loader :
+        A dynamic_loader which load a batch of data in an iterative manner. Dynamic loader batch output must be a
+        dictionary for BatchOutputFormat format. The dynamic loader must provide the data in a SHUFFLED way.
+    task_type : str
+        The task type of the data. can be one of the following: 'classification', 'semantic_segmentation',
+        'object_detection', 'other'. For 'other', only image validation checks will be run.
```

Comment:
Talk about what what will run (only image checks will run), not about what will be validated